packages/visx-tooltip/test/useTooltipInPortal.test.tsx

```diff
@@ -17,27 +19,77 @@ const TooltipWithZIndex = ({ zIndexOption, zIndexProp }: TooltipWithZIndexProps)
   return <TooltipInPortal zIndex={zIndexProp}>Hello</TooltipInPortal>;
 };
 
+interface TooltipWithPortalContaineProps {
+  shouldUsePortalContainer: boolean;
+}
+
+const TooltipWithPortalContainer = ({
+  shouldUsePortalContainer,
+}: TooltipWithPortalContaineProps) => {
+  // useState rather than useRef so it will react to the ref being available for testing purposes
+  const [portalContainer, setPortalContainer] = React.useState<HTMLDivElement | null>(null);
+  const onRefChange = React.useCallback((node) => {
+    setPortalContainer(node);
+  }, []);
+
+  const { TooltipInPortal } = useTooltipInPortal({
+    polyfill: ResizeObserver,
+    portalContainer: shouldUsePortalContainer ? portalContainer ?? undefined : undefined,
+  });
+
+  return (
+    <>
+      <div data-testid="inner-div" ref={shouldUsePortalContainer ? onRefChange : undefined} />
+      <TooltipInPortal>
+        <div data-testid="element-in-tooltip" />
+      </TooltipInPortal>
+    </>
+  );
+};
+
 describe('useTooltipInPortal()', () => {
   test('it should be defined', () => {
     expect(useTooltipInPortal).toBeDefined();
   });
 
-  it('should pass zIndex prop from options to Portal', () => {
-    const wrapper = shallow(<TooltipWithZIndex zIndexOption={1} />, {
-      disableLifecycleMethods: true,
-    }).dive();
-    const zIndex = wrapper.find('Portal').prop('zIndex');
-    expect(zIndex).toEqual(1);
+  describe('zIndex', () => {
+    it('should pass zIndex prop from options to Portal', () => {
+      const wrapper = shallow(<TooltipWithZIndex zIndexOption={1} />, {
+        disableLifecycleMethods: true,
+      }).dive();
+      const zIndex = wrapper.find('Portal').prop('zIndex');
+      expect(zIndex).toEqual(1);
+    });
+
+    it('should pass zIndex prop from component to Portal', () => {
+      const wrapper = shallow(
+        <TooltipWithZIndex zIndexOption={1} zIndexProp="var(--tooltip-zindex)" />,
+        {
+          disableLifecycleMethods: true,
+        },
+      ).dive();
+      const zIndex = wrapper.find('Portal').prop('zIndex');
+      expect(zIndex).toEqual('var(--tooltip-zindex)');
+    });
   });
 
-  it('should pass zIndex prop from component to Portal', () => {
-    const wrapper = shallow(
-      <TooltipWithZIndex zIndexOption={1} zIndexProp="var(--tooltip-zindex)" />,
-      {
-        disableLifecycleMethods: true,
-      },
-    ).dive();
-    const zIndex = wrapper.find('Portal').prop('zIndex');
-    expect(zIndex).toEqual('var(--tooltip-zindex)');
+  describe('portalContainer', () => {
+    it('should render tooltip at the document root', async () => {
+      render(<TooltipWithPortalContainer shouldUsePortalContainer={false} />);
+      const elementInPortal = await screen.findByTestId('element-in-tooltip');
+      expect(elementInPortal).toBeInTheDocument();
+      expect(
+        within(screen.getByTestId('inner-div')).queryByTestId('element-in-tooltip'),
```

Comment:
I guess same question on this test since `element-in-tooltip` is a sibling / wouldn't be expected to be a child of `inner-div`